components/server/src/authorization/relationship-updater.ts

```diff
@@ -39,6 +39,24 @@ export class RelationshipUpdater {
      * @returns
      */
     public async migrate(user: User): Promise<User> {
+        const isChecksEnabled = await isFgaChecksEnabled(user.id);
+        const migrated = this.internalMigrate(user);
+        // if checks are not enabled, we don't want to wait for the migration to finish
+        if (!isChecksEnabled) {
+            migrated.catch((err) => {
+                log.error({ userId: user.id }, "Error while migrating user", err);
+            });
+            return user;
+        }
+        try {
+            return await migrated;
+        } catch (error) {
+            log.error({ userId: user.id }, "Error while migrating user", error);
+            throw error;
+        }
+    }
+
+    private async internalMigrate(user: User): Promise<User> {
         const isEnabled = await isFgaWritesEnabled(user.id);
         if (!isEnabled) {
             if (user.additionalData?.fgaRelationshipsVersion !== undefined) {
```

Comment:
nit: would still love a explicit "Figration failed" catch block in internalMigrate.

Reply:
I added it to the migrate method